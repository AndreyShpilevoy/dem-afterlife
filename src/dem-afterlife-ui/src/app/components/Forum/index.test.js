/* eslint no-undef: 0, fp/no-unused-expression: 0, fp/no-nil: 0, fp/no-mutation:0 , react/jsx-filename-extension:0 */

import React from 'react';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {ForumPure} from './index';

configure({adapter: new Adapter()});

jest.mock('components/SubForum');

const classes = {
    '.default-Forum-lastTopicInfoWrapper>div': 'default-Forum-lastTopicInfoWrapper>div',
    '.default-Forum-mainContainer:last-child': 'default-Forum-mainContainer:last-child',
    '.default-Forum-mainContainer:last-child>.default-Forum-separator': 'default-Forum-mainContainer:last-child>.default-Forum-separator',
    bigText: 'default-Forum-bigText',
    center: 'default-Forum-center',
    centerMdUp: 'default-Forum-centerMdUp',
    disableRowOnSmXs: 'default-Forum-disableRowOnSmXs',
    displayInline: 'default-Forum-displayInline',
    lastTopicInfoWrapper: 'default-Forum-lastTopicInfoWrapper',
    mainContainer: 'default-Forum-mainContainer',
    separator: 'default-Forum-separator',
    smallText: 'default-Forum-smallText',
    subForumContainer: 'default-Forum-subForumContainer',
    topicsMessages: 'default-Forum-topicsMessages'
};

describe('ForumPure', () => {
    it('component match expected snapshot without subForumArray', () => {
        const forum = {
            id: 40,
            chapterId: 4,
            order: 4,
            title: 'Вопросы по созданию моделей',
            description: 'Вопросы по созданию моделей, конвертированию в игру и всего что с этим связано.',
            topicsCount: 4,
            postsCount: 864,
            lastTopicInfo: {
                topicId: 4,
                topicDescription: 'Exporter Ex Machina for Maya',
                latestPostTimeCreation: new Date('2007/09/06 22:33:09'),
                latestPostAuthorId: 4,
                latestPostAuthorName: 'Buba',
                latestPostAuthorGroupColor: '#00AA00'
            }
        };
        expect(shallow(<ForumPure className={'forumClassName'} classes={classes} forum={forum} />)).toMatchSnapshot();
    });

    it('component match expected snapshot with subForumArray', () => {
        const forum = {
            id: 40,
            chapterId: 4,
            order: 4,
            title: 'Вопросы по созданию моделей',
            description: 'Вопросы по созданию моделей, конвертированию в игру и всего что с этим связано.',
            topicsCount: 4,
            postsCount: 864,
            lastTopicInfo: {
                topicId: 4,
                topicDescription: 'Exporter Ex Machina for Maya',
                latestPostTimeCreation: new Date('2007/09/06 22:33:09'),
                latestPostAuthorId: 4,
                latestPostAuthorName: 'Buba',
                latestPostAuthorGroupColor: '#00AA00'
            },
            subForumArray: [
                {id: 51, title: 'FAQ', order: 57},
                {id: 52, title: 'Юзербары', order: 52},
                {id: 53, title: 'О игре', order: 53},
                {id: 54, title: 'Видео', order: 51},
                {id: 55, title: 'Рассказы', order: 55},
                {id: 56, title: 'Каталог файлов', order: 56}
            ]
        };
        expect(shallow(<ForumPure className={'forumClassName'} classes={classes} forum={forum} />)).toMatchSnapshot();
    });
});