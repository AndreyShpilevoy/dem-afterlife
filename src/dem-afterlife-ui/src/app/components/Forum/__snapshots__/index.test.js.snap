// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForumPure component match expected snapshot with subForumArray 1`] = `
<div
  className="default-Forum-mainContainer"
>
  <Jss(RowPure)
    className=""
    reverse={false}
  >
    <Jss(ColumnPure)
      className=""
      lg={9}
      lgOffset=""
      md={5}
      mdOffset=""
      sm=""
      smOffset=""
      xl=""
      xlOffset=""
      xs={12}
      xsOffset=""
    >
      <Jss(RowPure)
        className=""
        reverse={false}
      >
        <Jss(ColumnPure)
          className=""
          lg={8}
          lgOffset=""
          md=""
          mdOffset=""
          sm=""
          smOffset=""
          xl=""
          xlOffset=""
          xs={12}
          xsOffset=""
        >
          <Jss(LinkPure)
            className="default-Forum-bigText"
            to="/Forum/forumId=40"
          >
            Вопросы по созданию моделей
          </Jss(LinkPure)>
          <Jss(HiddenPure)
            className=""
            lg=""
            md="down"
            sm=""
            xl=""
            xs=""
          >
            <div
              className="default-Forum-smallText"
            >
              Вопросы по созданию моделей, конвертированию в игру и всего что с этим связано.
            </div>
          </Jss(HiddenPure)>
          <Jss(HiddenPure)
            className=""
            lg=""
            md="down"
            sm=""
            xl=""
            xs=""
          >
            <div
              className="default-Forum-subForumContainer"
            >
              <Jss(SubForumPure)
                key="51"
                subForum={
                  Object {
                    "id": 51,
                    "order": 57,
                    "title": "FAQ",
                  }
                }
              />
              <Jss(SubForumPure)
                key="52"
                subForum={
                  Object {
                    "id": 52,
                    "order": 52,
                    "title": "Юзербары",
                  }
                }
              />
              <Jss(SubForumPure)
                key="53"
                subForum={
                  Object {
                    "id": 53,
                    "order": 53,
                    "title": "О игре",
                  }
                }
              />
              <Jss(SubForumPure)
                key="54"
                subForum={
                  Object {
                    "id": 54,
                    "order": 51,
                    "title": "Видео",
                  }
                }
              />
              <Jss(SubForumPure)
                key="55"
                subForum={
                  Object {
                    "id": 55,
                    "order": 55,
                    "title": "Рассказы",
                  }
                }
              />
              <Jss(SubForumPure)
                key="56"
                subForum={
                  Object {
                    "id": 56,
                    "order": 56,
                    "title": "Каталог файлов",
                  }
                }
              />
            </div>
          </Jss(HiddenPure)>
        </Jss(ColumnPure)>
        <Jss(ColumnPure)
          className="default-Forum-center default-Forum-topicsMessages"
          lg={2}
          lgOffset=""
          md=""
          mdOffset=""
          sm=""
          smOffset=""
          xl=""
          xlOffset=""
          xs=""
          xsOffset=""
        >
          <Jss(HiddenPure)
            className=""
            lg="up"
            md=""
            sm=""
            xl=""
            xs=""
          >
            <Connect(Term)
              indentAfter={true}
              term={
                Object {
                  "id": 1,
                  "value": "Topics",
                }
              }
              untermedPostfix=":"
            />
          </Jss(HiddenPure)>
          4
        </Jss(ColumnPure)>
        <Jss(ColumnPure)
          className="default-Forum-center default-Forum-topicsMessages"
          lg={2}
          lgOffset=""
          md=""
          mdOffset=""
          sm=""
          smOffset=""
          xl=""
          xlOffset=""
          xs=""
          xsOffset=""
        >
          <Jss(HiddenPure)
            className=""
            lg="up"
            md=""
            sm=""
            xl=""
            xs=""
          >
            <Connect(Term)
              indentAfter={true}
              indentBefore={true}
              term={
                Object {
                  "id": 2,
                  "value": "Posts",
                }
              }
              untermedPostfix=":"
            />
          </Jss(HiddenPure)>
          864
        </Jss(ColumnPure)>
      </Jss(RowPure)>
    </Jss(ColumnPure)>
    <Jss(ColumnPure)
      className="default-Forum-centerMdUp"
      lg={3}
      lgOffset=""
      md={7}
      mdOffset=""
      sm=""
      smOffset=""
      xl=""
      xlOffset=""
      xs={12}
      xsOffset=""
    >
      <Jss(RowPure)
        className="default-Forum-disableRowOnSmXs"
        reverse={false}
      >
        <Jss(ColumnPure)
          className="default-Forum-center"
          lg={12}
          lgOffset=""
          md={6}
          mdOffset=""
          sm=""
          smOffset=""
          xl=""
          xlOffset=""
          xs=""
          xsOffset=""
        >
          <Jss(HiddenPure)
            className=""
            lg=""
            md=""
            sm="down"
            xl=""
            xs=""
          >
            <Jss(HiddenPure)
              className=""
              lg=""
              md="down"
              sm=""
              xl=""
              xs=""
            >
              <UserName
                className=""
                color="#00AA00"
                id={4}
                name="Buba"
              />
              <Connect(Term)
                doNotTerm={true}
                indentAfter={true}
                indentBefore={true}
                spaceAfter={true}
                spaceBefore={true}
                untermedPostfix="-"
              />
            </Jss(HiddenPure)>
            <Connect(RelativeDateTime)
              relativeDateTime={2007-09-06T19:33:09.000Z}
              spaceBefore={true}
            />
          </Jss(HiddenPure)>
        </Jss(ColumnPure)>
        <Jss(ColumnPure)
          className="default-Forum-lastTopicInfoWrapper"
          lg={12}
          lgOffset=""
          md={6}
          mdOffset=""
          sm=""
          smOffset=""
          xl=""
          xlOffset=""
          xs=""
          xsOffset=""
        >
          <div>
            <Jss(HiddenPure)
              className=""
              lg="up"
              md=""
              sm=""
              xl=""
              xs=""
            >
              <Connect(Term)
                indentAfter={true}
                term={
                  Object {
                    "id": 3,
                    "value": "Last message",
                  }
                }
                untermedPostfix=":"
              />
            </Jss(HiddenPure)>
            <Jss(LinkPure)
              className="default-Forum-smallText default-Forum-displayInline"
              to="/Topic/topicId=4"
            >
              Exporter Ex Machina for Maya
            </Jss(LinkPure)>
          </div>
        </Jss(ColumnPure)>
      </Jss(RowPure)>
    </Jss(ColumnPure)>
  </Jss(RowPure)>
  <div
    className="default-Forum-separator"
  />
</div>
`;

exports[`ForumPure component match expected snapshot without subForumArray 1`] = `
<div
  className="default-Forum-mainContainer"
>
  <Jss(RowPure)
    className=""
    reverse={false}
  >
    <Jss(ColumnPure)
      className=""
      lg={9}
      lgOffset=""
      md={5}
      mdOffset=""
      sm=""
      smOffset=""
      xl=""
      xlOffset=""
      xs={12}
      xsOffset=""
    >
      <Jss(RowPure)
        className=""
        reverse={false}
      >
        <Jss(ColumnPure)
          className=""
          lg={8}
          lgOffset=""
          md=""
          mdOffset=""
          sm=""
          smOffset=""
          xl=""
          xlOffset=""
          xs={12}
          xsOffset=""
        >
          <Jss(LinkPure)
            className="default-Forum-bigText"
            to="/Forum/forumId=40"
          >
            Вопросы по созданию моделей
          </Jss(LinkPure)>
          <Jss(HiddenPure)
            className=""
            lg=""
            md="down"
            sm=""
            xl=""
            xs=""
          >
            <div
              className="default-Forum-smallText"
            >
              Вопросы по созданию моделей, конвертированию в игру и всего что с этим связано.
            </div>
          </Jss(HiddenPure)>
          <Jss(HiddenPure)
            className=""
            lg=""
            md="down"
            sm=""
            xl=""
            xs=""
          />
        </Jss(ColumnPure)>
        <Jss(ColumnPure)
          className="default-Forum-center default-Forum-topicsMessages"
          lg={2}
          lgOffset=""
          md=""
          mdOffset=""
          sm=""
          smOffset=""
          xl=""
          xlOffset=""
          xs=""
          xsOffset=""
        >
          <Jss(HiddenPure)
            className=""
            lg="up"
            md=""
            sm=""
            xl=""
            xs=""
          >
            <Connect(Term)
              indentAfter={true}
              term={
                Object {
                  "id": 1,
                  "value": "Topics",
                }
              }
              untermedPostfix=":"
            />
          </Jss(HiddenPure)>
          4
        </Jss(ColumnPure)>
        <Jss(ColumnPure)
          className="default-Forum-center default-Forum-topicsMessages"
          lg={2}
          lgOffset=""
          md=""
          mdOffset=""
          sm=""
          smOffset=""
          xl=""
          xlOffset=""
          xs=""
          xsOffset=""
        >
          <Jss(HiddenPure)
            className=""
            lg="up"
            md=""
            sm=""
            xl=""
            xs=""
          >
            <Connect(Term)
              indentAfter={true}
              indentBefore={true}
              term={
                Object {
                  "id": 2,
                  "value": "Posts",
                }
              }
              untermedPostfix=":"
            />
          </Jss(HiddenPure)>
          864
        </Jss(ColumnPure)>
      </Jss(RowPure)>
    </Jss(ColumnPure)>
    <Jss(ColumnPure)
      className="default-Forum-centerMdUp"
      lg={3}
      lgOffset=""
      md={7}
      mdOffset=""
      sm=""
      smOffset=""
      xl=""
      xlOffset=""
      xs={12}
      xsOffset=""
    >
      <Jss(RowPure)
        className="default-Forum-disableRowOnSmXs"
        reverse={false}
      >
        <Jss(ColumnPure)
          className="default-Forum-center"
          lg={12}
          lgOffset=""
          md={6}
          mdOffset=""
          sm=""
          smOffset=""
          xl=""
          xlOffset=""
          xs=""
          xsOffset=""
        >
          <Jss(HiddenPure)
            className=""
            lg=""
            md=""
            sm="down"
            xl=""
            xs=""
          >
            <Jss(HiddenPure)
              className=""
              lg=""
              md="down"
              sm=""
              xl=""
              xs=""
            >
              <UserName
                className=""
                color="#00AA00"
                id={4}
                name="Buba"
              />
              <Connect(Term)
                doNotTerm={true}
                indentAfter={true}
                indentBefore={true}
                spaceAfter={true}
                spaceBefore={true}
                untermedPostfix="-"
              />
            </Jss(HiddenPure)>
            <Connect(RelativeDateTime)
              relativeDateTime={2007-09-06T19:33:09.000Z}
              spaceBefore={true}
            />
          </Jss(HiddenPure)>
        </Jss(ColumnPure)>
        <Jss(ColumnPure)
          className="default-Forum-lastTopicInfoWrapper"
          lg={12}
          lgOffset=""
          md={6}
          mdOffset=""
          sm=""
          smOffset=""
          xl=""
          xlOffset=""
          xs=""
          xsOffset=""
        >
          <div>
            <Jss(HiddenPure)
              className=""
              lg="up"
              md=""
              sm=""
              xl=""
              xs=""
            >
              <Connect(Term)
                indentAfter={true}
                term={
                  Object {
                    "id": 3,
                    "value": "Last message",
                  }
                }
                untermedPostfix=":"
              />
            </Jss(HiddenPure)>
            <Jss(LinkPure)
              className="default-Forum-smallText default-Forum-displayInline"
              to="/Topic/topicId=4"
            >
              Exporter Ex Machina for Maya
            </Jss(LinkPure)>
          </div>
        </Jss(ColumnPure)>
      </Jss(RowPure)>
    </Jss(ColumnPure)>
  </Jss(RowPure)>
  <div
    className="default-Forum-separator"
  />
</div>
`;
